{
    "CVE-ID": "CVE-2008-1887",
    "NVD": {
        "Descriptions": "Python 2.5.2 and earlier allows context-dependent attackers to execute arbitrary code via multiple vectors that cause a negative size value to be provided to the PyString_FromStringAndSize function, which allocates less memory than expected when assert() is disabled and triggers a buffer overflow.",
        "Cvss2_BaseScore": 9.3,
        "Cvss3_BaseScore": 0,
        "Cvss2_VectorString": "AV:N/AC:M/Au:N/C:C/I:C/A:C",
        "Cvss3_VectorString": "",
        "CPEs": [
            {
                "URI": "cpe:/o:debian:debian_linux:4.0",
                "FormattedString": "cpe:2.3:o:debian:debian_linux:4.0:*:*:*:*:*:*:*",
                "VersionStartExcluding": "",
                "VersionStartIncluding": "",
                "VersionEndExcluding": "",
                "VersionEndIncluding": ""
            },
            {
                "URI": "cpe:/o:canonical:ubuntu_linux:8.04::~~-~~~",
                "FormattedString": "cpe:2.3:o:canonical:ubuntu_linux:8.04:*:*:*:-:*:*:*",
                "VersionStartExcluding": "",
                "VersionStartIncluding": "",
                "VersionEndExcluding": "",
                "VersionEndIncluding": ""
            },
            {
                "URI": "cpe:/o:canonical:ubuntu_linux:7.10",
                "FormattedString": "cpe:2.3:o:canonical:ubuntu_linux:7.10:*:*:*:*:*:*:*",
                "VersionStartExcluding": "",
                "VersionStartIncluding": "",
                "VersionEndExcluding": "",
                "VersionEndIncluding": ""
            },
            {
                "URI": "cpe:/o:canonical:ubuntu_linux:7.04",
                "FormattedString": "cpe:2.3:o:canonical:ubuntu_linux:7.04:*:*:*:*:*:*:*",
                "VersionStartExcluding": "",
                "VersionStartIncluding": "",
                "VersionEndExcluding": "",
                "VersionEndIncluding": ""
            },
            {
                "URI": "cpe:/o:canonical:ubuntu_linux:6.06",
                "FormattedString": "cpe:2.3:o:canonical:ubuntu_linux:6.06:*:*:*:*:*:*:*",
                "VersionStartExcluding": "",
                "VersionStartIncluding": "",
                "VersionEndExcluding": "",
                "VersionEndIncluding": ""
            },
            {
                "URI": "cpe:/a:python:python",
                "FormattedString": "cpe:2.3:a:python:python:*:*:*:*:*:*:*:*",
                "VersionStartExcluding": "",
                "VersionStartIncluding": "",
                "VersionEndExcluding": "",
                "VersionEndIncluding": "2.5.2"
            }
        ]
    },
    "Official": {
        "2.4.6c1": {
            "source": [
                "https://github.com/python/cpython/blob/2.4/Misc/NEWS"
            ],
            "BTS": {
                "bpo2587": {
                    "ID": "bpo2587",
                    "url": "https://bugs.python.org/issue2587",
                    "date": "",
                    "nonce": "",
                    "section": ""
                }
            }
        },
        "2.5.3c1": {
            "source": [
                "https://github.com/python/cpython/blob/2.5/Misc/NEWS"
            ],
            "BTS": {
                "bpo2587": {
                    "ID": "bpo2587",
                    "url": "https://bugs.python.org/issue2587",
                    "date": "",
                    "nonce": "",
                    "section": ""
                }
            }
        }
    }
}